bycycle/plts/burst.py

```diff
@@ -0,0 +1,227 @@
+        for _, cyc in df.iterrows():
+
+            if cyc[column] < osc_kwargs[osc_key]:
+                axes[0].axvspan(times[cyc['sample_last_' + side_e]],
+                                times[cyc['sample_next_' + side_e]],
+                                alpha=0.5, color=color, lw=0)
+
+        # Plot each burst param on separate axes
+        if not plot_only_result:
+
+            ylabel = osc_key.replace('_threshold', '').replace('_', ' ').capitalize()
+            xlabel = 'Time (s)' if idx == len(burst_params)-1 else ''
+
+            plot_burst_detect_param(df, sig, fs, burst_params[idx], osc_kwargs[osc_key],
+                                    figsize=figsize, ax=axes[idx+1], tlims=tlims,
+                                    xlabel=xlabel, ylabel=ylabel, color=color, interp=interp)
+
+
+def plot_burst_detect_param(df, sig, fs, burst_param, thresh, tlims=None,
+                            ax=None, interp=True, **kwargs):
+    """Plot a burst detection parameter and threshold.
+
+    Parameters
+    ----------
+    df : pandas.DataFrame
+        Dataframe output of :func:`~.compute_features`.
+    sig : 1d array
+        Time series to plot.
+    fs : float
+        Sampling rate, in Hz.
+    burst_param : str
+        Column name of the parameter of interest in ``df``.
+    thresh : float
+        The burst parameter threshold. Parameter values greater
+        than ``thresh`` are considered bursts.
+    tlims : tuple of (float, float), optional, default: None
+        Start and stop times for plot.
+    ax : matplotlib.Axes, optional
+        Figure axes upon which to plot.
+    interp : bool
+        Interpolates points if true.
+    **kwargs
+        Keyword arguments to pass into `plot_time_series`.
+
+    Notes
+    -----
+    Default keyword arguments include:
+
+    - ``figsize``: tuple of (float, float), default: (15, 3)
+    - ``xlabel``: str, default: 'Time (s)'
+    - ``ylabel``: str, default: 'Voltage (uV)
+    - ``color``: str, default: 'r'.
+
+      - Note: ``color`` here is the fill color, rather than line color.
+
+    """
+
+    # Set default kwargs
+    figsize = kwargs.pop('figsize', (15, 3))
+    xlabel = kwargs.pop('xlabel', 'Time (s)')
+    ylabel = kwargs.pop('ylabel', burst_param)
+    color = kwargs.pop('color', 'r')
+
+    # Determine time array and limits
+    times = np.arange(0, len(sig) / fs, 1 / fs)
+    tlims = (times[0], times[-1]) if tlims is None else tlims
+
+    if ax is None:
+        fig, ax = plt.subplots(figsize=figsize)
+
+    # Determine extrema strings
+    center_e, side_e = get_extrema(df)
+
+    df, sig, times = apply_tlims(df, sig, times, fs, tlims)
+
+    # Plot burst param
+    if interp:
+
+        plot_time_series([times[df['sample_' + center_e]], tlims], [df[burst_param], [thresh]*2],
+                         ax=ax, colors=['k.-', 'k--'], xlim=tlims, xlabel=xlabel,
+                         ylabel="{0:s}\nthreshold={1:.2f}".format(ylabel, thresh), **kwargs)
+
+    else:
```

Comment:
I think perhaps a little more incode comments here could be useful to briefly describe what is happening.